packages/aws-cdk-lib/aws-iam/lib/role.ts

```diff
@@ -375,7 +375,7 @@ export class Role extends Resource implements IRole {
   /**
    * The assume role policy document associated with this role.
    */
-  public readonly assumeRolePolicy?: PolicyDocument;
+  public readonly assumeRolePolicy: PolicyDocument;
```

Comment:
we cannot make an optional props to be required as it is a breaking change and can lead to failures for multiple users.

Reply:
This seems to be **public readonly** property. and this is set in constructor (see below) and moreover this is the **only place we set this**. There is no way user will set this. and There is no way it will be undefined, based on how it is being used in constructor IMO. So, ideally it shouldn't break anything. it should be backward compatible.

https://github.com/aws/aws-cdk/blob/4e7d4804093d1a2a54f03830a3770d56c4729bc8/packages/aws-cdk-lib/aws-iam/lib/role.ts#L425

please let me know does this makes sense.